configfile: "config.yaml"
sample= config["sample"]


rule all:
    input:
        expand("{sample}_HTSeq.csv", sample=config["sample"])



rule hisat2_allign:
    input:
        reads=["{sample}_1.fastq", "{sample}_2.fastq"]
    output:
        "{sample}.sam"
    log:
        "logs/hisat2_align_{sample}.log"
    params:
        extra="",
        idx="grch38/genome"
    threads: 2
    shell:
        "hisat2 -x {params.idx} -1 SRR10777215_1.fastq -2 SRR10777215_2.fastq -S {output}"

        #"wget https://genome-idx.s3.amazonaws.com/hisat/grch38_genome.tar.gz \
# -1 {input.fastq1} -2 {input.fastq2}



rule htseq:
        input:
            "{sample}.bam"
        output:
            '{sample}_HTSeq_union_gff3_no_gene_ID.log',
            '{sample}_HTSeq.csv'
        threads: 1
        shell:
            'htseq-count -m union -s no -t gene -i ID -r pos -f bam {input.bam} {input.gff} &> {output[0]} && '
            'grep ENS {output[0]} | sed "s/gene://g" > {output[1]}'
